trace/trace.go

```diff
@@ -123,6 +123,41 @@ type StartOptions struct {
 	// SpanKind represents the kind of a span. If none is set,
 	// SpanKindUnspecified is used.
 	SpanKind int
+
+	parent       *SpanContext
+	parentRemote bool
+}
+
+type StartSpanOption func(*StartOptions)
+
+func WithSpanKind(spanKind int) StartSpanOption {
+	return func(o *StartOptions) {
+		o.SpanKind = spanKind
+	}
+}
+
+func WithSampler(sampler Sampler) StartSpanOption {
+	return func(o *StartOptions) {
+		o.Sampler = sampler
+	}
+}
+
+func WithRemoteParent(parent SpanContext) StartSpanOption {
+	return func(o *StartOptions) {
+		o.parent = &parent
+		o.parentRemote = true
+	}
+}
+
+func WithParent(span *Span) StartSpanOption {
```

Comment:
I don't think we should ever do this. Parent should always determined from the incoming context if it is not remote.

Reply:
This was mostly added to *avoid* taking the parent from the context e.g. in ocgrpc, ochttp we take a Context but we want to ignore the Span in that context since we are starting a new trace. I will change it to something like "AsRootSpan()" which just sets parent to nil.

Reply:
Ah I see. I actually would prefer us providing an option that makes the span root instead.

Reply:
actually realized I can just explicitly clear it from the context